src/main/kotlin/net/irisshaders/lilybot/extensions/util/ImageChannel.kt

```diff
@@ -0,0 +1,159 @@
+package net.irisshaders.lilybot.extensions.util
+
+import com.kotlindiscord.kord.extensions.checks.anyGuild
+import com.kotlindiscord.kord.extensions.checks.guildFor
+import com.kotlindiscord.kord.extensions.checks.hasPermission
+import com.kotlindiscord.kord.extensions.commands.application.slash.ephemeralSubCommand
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.extensions.ephemeralSlashCommand
+import com.kotlindiscord.kord.extensions.extensions.event
+import com.kotlindiscord.kord.extensions.types.respond
+import com.kotlindiscord.kord.extensions.utils.delete
+import com.kotlindiscord.kord.extensions.utils.isNullOrBot
+import com.kotlindiscord.kord.extensions.utils.respond
+import com.soywiz.klock.seconds
+import dev.kord.common.entity.Permission
+import dev.kord.core.event.message.MessageCreateEvent
+import dev.kord.core.exception.EntityNotFoundException
+import dev.kord.rest.builder.message.create.embed
+import kotlinx.coroutines.delay
+import net.irisshaders.lilybot.utils.DatabaseHelper.deleteImageChannel
+import net.irisshaders.lilybot.utils.DatabaseHelper.getImageChannels
+import net.irisshaders.lilybot.utils.DatabaseHelper.setImageChannel
+import net.irisshaders.lilybot.utils.configPresent
+
+/**
+ * The class the holds the systems that allow a guild to set a channel as image only.
+ *
+ * @since 3.3.0
+ */
+class ImageChannel : Extension() {
+	override val name = "imagechannel"
+
+	override suspend fun setup() {
+		/**
+		 * Image channel commands.
+		 * @author NoComment1105
+		 * @since 3.3.0
+		 */
+		ephemeralSlashCommand {
+			name = "image-channel"
+			description = "The parent command for image channel setting"
+
+			/**
+			 * The command that sets the image channel.
+			 */
+			ephemeralSubCommand {
+				name = "set"
+				description = "Set a channel as an image channel"
+
+				check {
+					anyGuild()
+					configPresent()
+					hasPermission(Permission.ManageGuild)
+				}
+
+				action {
+					setImageChannel(guild!!.id, channel.asChannel().id)
+
+					respond {
+						content = "Set channel as image only"
+					}
+				}
+			}
+
+			/**
+			 * The command that unsets the image channel.
+			 */
+			ephemeralSubCommand {
+				name = "unset"
+				description = "Unset a channel as an image channel"
+
+				check {
+					anyGuild()
+					configPresent()
+					hasPermission(Permission.ManageGuild)
+				}
+
+				action {
+					deleteImageChannel(guild!!.id, channel.asChannel().id)
+
+					respond {
+						content = "Unset channel as image only"
```

Comment:
This should check if it's already an image channel and respond accordingly. Same for setting. Because of this, I can create a bug where the same channel is added to the database many times.
<img width="615" alt="Screen Shot 2022-05-29 at 10 03 49 PM" src="https://user-images.githubusercontent.com/66021850/170904045-d1b7fbd3-0467-40c3-b8bb-c6600893d278.png">
e